helix-core/src/main/java/org/apache/helix/util/HelixUtil.java

```diff
@@ -533,21 +533,38 @@ public static long getSystemPropertyAsLong(String propertyKey, long propertyDefa
 
   /**
    * Compose the config for an instance
-   * @param instanceName
+   * @param instanceName the unique name of the instance
    * @return InstanceConfig
    */
+  @Deprecated
   public static InstanceConfig composeInstanceConfig(String instanceName) {
-    InstanceConfig instanceConfig = new InstanceConfig(instanceName);
-    String hostName = instanceName;
-    String port = "";
+    return composeInstanceConfig(instanceName, new InstanceConfig.Builder());
+  }
+
+  /**
+   * Compose the config for an instance with defaults if provided.
+   * @param instanceName the unique name of the instance
+   * @param defaultInstanceConfigBuilder instance config builder filled with defaults
+   * @return InstanceConfig
+   */
+  public static InstanceConfig composeInstanceConfig(String instanceName,
```

Comment:
If we already have Builder, would assume the parsing logic is already done and it is overlap with build fields... Let's not have them together.

So I think we can keep the composeInstanceConfig for previous one. And the build itself is already the logic to generate the instance config. Not need to have facility function for it.